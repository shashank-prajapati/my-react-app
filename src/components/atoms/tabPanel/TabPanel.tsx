import { Box } from "@mui/material";
import React from "react";

interface TabPanelProps {
  children?: React.ReactNode;
  index: number;
  value: number;
}

function TabPanel(props: TabPanelProps) {
  const { children, value, index, ...other } = props;

  return (
    <div
      role="tabpanel"
      hidden={value !== index}
      id={`simple-tabpanel-${index}`}
      aria-labelledby={`simple-tab-${index}`}
      {...other}
    >
      {value === index && <Box sx={{pt:5}}>{children}</Box>}
    </div>
  );
}
export default TabPanel;
