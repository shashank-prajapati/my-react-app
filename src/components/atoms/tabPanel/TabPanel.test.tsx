import * as React from 'react';
import { render, fireEvent } from '@testing-library/react';
import TabPanel from './TabPanel';
import { Typography } from '@mui/material';


describe('Testing Tab Panel', ()=>{
    it('should match the default snapshot',()=>{
        const {container} = render(<TabPanel index={0} value={0}><Typography>Hello World</Typography></TabPanel>,);
        expect(container).toMatchSnapshot();
    })
})

