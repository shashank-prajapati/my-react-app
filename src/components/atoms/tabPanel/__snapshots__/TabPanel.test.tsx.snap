// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Tab Panel should match the default snapshot 1`] = `
<div>
  <div
    aria-labelledby="simple-tab-0"
    id="simple-tabpanel-0"
    role="tabpanel"
  >
    <div
      class="MuiBox-root css-pxpt32"
    >
      <p
        class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
      >
        Hello World
      </p>
    </div>
  </div>
</div>
`;
